src/dataflow/source/kafka.rs

```diff
@@ -134,29 +167,36 @@ fn refresh_kafka_metadata(
 fn create_kafka_consumer(
     activator: Arc<Mutex<SyncActivator>>,
     topic: &str,
+    partition_id: i32,
     kafka_config: &ClientConfig,
 ) -> BaseConsumer<GlueConsumerContext> {
     let cx = GlueConsumerContext(activator);
     let consumer: BaseConsumer<GlueConsumerContext> = kafka_config
         .create_with_context(cx)
         .expect("Failed to create Kafka Consumer");
-    consumer.subscribe(&[topic]).unwrap();
+    let mut partition_list = TopicPartitionList::with_capacity(1);
+    partition_list.add_partition(topic, partition_id);
+    consumer.assign(&partition_list).unwrap();
     consumer
 }
 
 /// Update the list of Kafka consumers to match the number of partitions
 /// We currently create one consumer per partition
 fn update_consumer_list(
+    current_max_pid: i32,
     consumers: &mut VecDeque<BaseConsumer<GlueConsumerContext>>,
     topic: &str,
-    to_add: usize,
+    to_add: i32,
```

Comment:
you could keep this as usize and cast it to i32 in the body of the function so the arithmetic typechecks. Would it ever make sense to pass a negative number in?